File: leetcode/1. Two Sum/Hash Map, One pass/hash_map_one_pass.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		if secondIndex, ok := m[complement]; ok {
			return []int{i, secondIndex}
		}
		m[num] = i
	}
	return []int{}
}

File: leetcode/1. Two Sum/Hash Map, One pass/hash_map_one_pass.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        for(int i=0; i<nums.size(); i++){
            int complement = target-nums[i];
            if(m.count(complement)>0){
                return vector<int>{i, m[complement]};
            }
            m[nums[i]] = i;
        }
        return vector<int>{};
    }
};